src/Controller/CRUDController.php

```diff
@@ -1558,7 +1558,7 @@ protected function preList(Request $request)
     protected function handleXmlHttpRequestErrorResponse(Request $request, FormInterface $form): ?JsonResponse
     {
         if (empty(array_intersect(['application/json', '*/*'], $request->getAcceptableContentTypes()))) {
-            @trigger_error('In next major version response will return 406 NOT ACCEPTABLE without `Accept: application/json` or `Accept: */*`', E_USER_DEPRECATED);
+            @trigger_error(sprintf('%s is not supported since sonata-project/admin-bundle 3.x and will return 406 in 4.0. Use `application/json`.', implode(',', $request->getAcceptableContentTypes())));
```

Comment:
```suggestion
            @trigger_error(sprintf('%s is not supported since sonata-project/admin-bundle 3.x and will return 406 in 4.0. Use `application/json`.', implode(',', $request->getAcceptableContentTypes())));
```
```suggestion
            @trigger_error(sprintf('"%s" is not supported since sonata-project/admin-bundle 3.x and will return 406 in 4.0. Use `application/json`.', implode('", "', $request->getAcceptableContentTypes())));
```
IMO, this message lacks the context (request, XHR, etc) . If this is shown in the deprecation logs, I think it will be difficult to understand where it was triggered and what should be done to fix the problem.

Reply:
BTW, we should also be more explicit about this message:
>will return 406

 I propose something like:
>the status code 406 will be set in the returned response